vignettes/slopes.Rmd

```diff
@@ -15,103 +14,390 @@ knitr::opts_chunk$set(
+library(tmap)
+tmap_mode("view")
+tm_shape(sf_linestring) +
+  tm_lines(lwd = 5) +
+  tm_basemap(leaflet::providers$Esri.WorldTopoMap)
+```
+
+Imagine you want to calculate the gradient of the route shown above. 
+You can do this as a two step process as follows.
 
-```{r cumulative-slopes, fig.cap="Illustration of the importance of slope length. 4 segments with an 8% gradient is not the same as a single segment with a gradient of 8%.", out.width="40%", echo=FALSE}
-knitr::include_graphics("SLOPES-commulative-slope-1.png")
+**Step 1**: add elevations to each coordinate in the linestring (requires a [MapBox API](https://account.mapbox.com/access-tokens/) key):
+
+```{r}
+sf_linestring_xyz = elevation_add(sf_linestring)
 ```
 
-This is accounted for in slope calculation methods that take the distance-weighted mean of slopes.
+You can check the elevations added to the new `sf_linestring_xyz` object by printing its coordinates, as follows (note the new Z column that goes from above 87 m above sea level to only 79 m in a short distance).
 
 ```{r}
+st_coordinates(sf_linestring_xyz)
+```
+
+You can use the `z_` functions to extract such values:
+
+```{r}
+z_value(sf_linestring_xyz) # returns all the elevation values between xy coordinates
+
+z_mean(sf_linestring_xyz) # elevation mean value
+z_min(sf_linestring_xyz) # elevation min value 
+z_max(sf_linestring_xyz) # elevation max value 
+z_start(sf_linestring_xyz) # first z
+z_end(sf_linestring_xyz) # last z
+```
+
+
+**Step 2**: calculate the average slope of the linestring
+
+```{r}
+slope_xyz(sf_linestring_xyz)
+```
+
+The result, just over 0.2, tells us that it's quite a steep slope: a 21% gradient *on average*.
+
+### Route + available DEM
+
+Using the slopes package we can estimate the gradient of individual road segments. When these segments are combined into routes, we then need a means of assessing the hilliness of the entire route. A range of indices can be used to represent route hilliness. The choice of which index is most appropriate may be context dependent (see the [introducion to slopes](https://itsleeds.github.io/intro-to-slopes/) vignette).
+  
+Again, let us use the same function with a entire route, `lisbon_route`, also available in the package:
+
+```{r message=FALSE, warning=FALSE}
+sf_route = lisbon_route
+class(sf_route)
+st_geometry_type(sf_route)
+
+tm_shape(sf_route) +
+  tm_lines(lwd = 3) +
+  tm_basemap(leaflet::providers$Esri.WorldTopoMap)
+```
+
+**Step 1**: add elevations to each coordinate in the route:
+
+```{r}
+sf_route_xyz = elevation_add(sf_route)
+```
+
+**Step 2**: calculate the average slope of the route
+
+```{r}
+slope_xyz(sf_route_xyz)
+```
+
+The result shows a 7.7% gradient *on average*.
+
+Now, if you already have a DEM, you can calculate the slopes directly as follows, with `slope_raster()`:
+
+```{r}
+class(dem_lisbon_raster)
+slope_raster(routes = sf_route,
+             dem = dem_lisbon_raster)
+```
+The result shows a 7.8% gradient *on average*.
```

Comment:
Great example linking it back to another steep route, on that is composed of many segments.